src/test/java/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchAPIIntegrationTest.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.justice.hmpps.offendersearch.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.restassured.RestAssured;
+import io.restassured.config.ObjectMapperConfig;
+import io.restassured.config.RestAssuredConfig;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestContext;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.AbstractTestExecutionListener;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import uk.gov.justice.hmpps.offendersearch.dto.OffenderDetail;
+import uk.gov.justice.hmpps.offendersearch.util.LocalStackHelper;
+
+import java.io.IOException;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("dev,localstack")
+@RunWith(SpringJUnit4ClassRunner.class)
+@TestExecutionListeners(listeners = {
+        DependencyInjectionTestExecutionListener.class,
+        OffenderSearchAPIIntegrationTest.class})
+@ContextConfiguration
+public class OffenderSearchAPIIntegrationTest extends AbstractTestExecutionListener {
+
+    @LocalServerPort
+    int port;
+
+    @Value("${test.token.good}")
+    private String validOauthToken;
+
+    @Override
+    public void beforeTestClass(TestContext testContext) throws IOException {
```

Comment:
I needed to initialise elasticsearch once for the test class, this got fiddly as @BeforeClass requires a static method and I needed some injected beans and values. I've gone with this approach but there is probably a better way

Reply:
suggested using @before with a static flag